src/traversals/greedy_color.jl

```diff
@@ -0,0 +1,124 @@
+"""
+    struct coloring{T}
+
+Stores number of colors used and mapping from vertex to color
+"""
+
+struct coloring{T<:Integer}
+    num_colors::T
+    colors::Vector{T}
+end
+
+best_color(c1::coloring, c2::coloring) = c1.num_colors < c2.num_colors ? c1 : c2
+
+@doc_str """
+    Perform greedy coloring [https://en.wikipedia.org/wiki/Greedy_coloring]
```

Comment:
Needs the function signature as the first line in the docstring, indented. The description is not indented:
```
"""
    GCD(a, b)

Return the greatest common denominator of `a` and `b`.
"""
function GCD(a, b)
  ...
end
```